src/Tools/dotnet-monitor/CollectionRules/IAction.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules
+{
+    internal interface IAction<TOptions>
+    {
+        Task<ActionResponse> Execute(TOptions options, CancellationToken token);
+    }
+
+    internal class ActionResponse
```

Comment:
nit:

```suggestion
    internal struct ActionResult
```

Regarding rename: When I think of responses, I also think of there being a request. But there is no structural notion of a request.

Regarding class vs struct: I don't think anything is ever going to derive from this type and this type is likely only going to be used in locals (never saved off to the heap), so it's probably better for this to be a struct. @wiktork, what do you think?

Reply:
Yeah, I agree, maybe `CollectionRuleActionResult` to better differentiate.